### DISCUSSION TITLE
post/060224-powerautomate-odata-filter-date/

### DISCUSSION THREAD

**Discussion by giscus** [AlexanderHenkel/blogComments#3] [Announcements]
# post/060224-powerautomate-odata-filter-date/

Learn how to use the ODATA to filter on date columns in SharePoint, and how to format your date

https://alexanderhenkel.dk/post/060224-powerautomate-odata-filter-date/

**Comment by JeremyTBradshaw** [comment]
I was excited when reading this but when trying it with the Get Events V4 action in Power Automate, I get the following:

```
"message": "Invalid filter clause: A binary operator with incompatible types was detected. Found operand types 'microsoft.graph.dateTimeTimeZone' and 'Edm.String' for operator kind 'GreaterThanOrEqual'.",
            "code": "BadRequest",
```

My query is:
```
"$filter": "isOrganizer eq true and Start ge '2025-05-23T00:00:01Z' and Start le '2025-05-23T23:59:59Z'"
```
I can't attach a screenshot but I tried with and without the quotes, and the dates are actually entered using the formatDateTime expressions (the purple fx's), exactly how your example is.  Wondering if you know a way around this?

**Reply by AlexanderHenkel** [reply]
You need to replace **start** with **start/DateTime**

```"$filter": "isOrganizer eq true and start/DateTime ge '2025-05-23T00:00:01Z' and start/DateTime le '2025-05-23T23:59:59Z'"```

**Reply by JeremyTBradshaw** [reply]
Thanks very much. I also eventually learned that earlier too and was able to get it done.  Next task is to parse the attendees and do stuff with them.
